tasks/cockpit-tasks

```diff
@@ -113,19 +113,41 @@ function run_one_task() {
     fi
 }
 
+function run_from_queue() {
+    cd "$XDG_CACHE_HOME"/cockpit-project/bots
+    # Consuming directly from queue, commands generated by run-queue,
+    # tests-scan, and issue-scan have bots/ prepended
+    [ -e bots ] || ln -s . bots
+    if /usr/bin/timeout 12h /bin/sh -exc "./run-queue --amqp \"$AMQP_SERVER\""; then
+        work_done=1
+    fi
+}
+
 # on mass deployment, avoid GitHub stampede
 sleep $(shuf -i 0-120 -n 1)
 
-# Perform a couple of tasks or waits, then restart
-for i in $(seq 1 30); do
-    work_done=
+PROJECTS="
+cockpit-project/bots
+cockpit-project/cockpit
+cockpit-project/starter-kit
+cockpit-project/cockpit-podman
+weldr/cockpit-composer
+"
 
-    update_repo  "$XDG_CACHE_HOME"/cockpit-project/bots             https://github.com/cockpit-project/cockpit
-    run_one_task "$XDG_CACHE_HOME"/cockpit-project/cockpit          https://github.com/cockpit-project/cockpit
-    run_one_task "$XDG_CACHE_HOME"/cockpit-project/starter-kit      https://github.com/cockpit-project/starter-kit
-    run_one_task "$XDG_CACHE_HOME"/weldr/cockpit-composer           https://github.com/weldr/cockpit-composer
-    run_one_task "$XDG_CACHE_HOME"/cockpit-project/cockpit-podman   https://github.com/cockpit-project/cockpit-podman
+# check for tasks
+for P in $PROJECTS; do
+    work_done=
+    update_repo  "$XDG_CACHE_HOME"/cockpit-project/bots https://github.com/cockpit-project/bots
+    run_one_task "$XDG_CACHE_HOME"/"$P"                 https://github.com/"$P"
+    # Nothing to do, or failure
+    [ -n "$work_done" ] || slumber
```

Comment:
Also, we don't need to sleep between these IMHO -- we didn't so far either, and it didn't hurt.